.github/workflows/builds.yml

```diff
@@ -24,7 +24,7 @@ jobs:
         uses: actions/setup-python@v5
         with:
           # MacOS issue ref.: https://github.com/actions/setup-python/issues/855 & https://github.com/actions/setup-python/issues/865
-          python-version: ${{ matrix.os == 'macos-latest' && '3.11' || matrix.python }}
+          python-version: ${{ matrix.os == 'macos-latest' && matrix.python == '3.10' && '3.11' || matrix.python }}
```

Comment:
Why is there `3.11` written in this condition ?

Reply:
It's still the action bug with macOS for the python matrix == 3.10 uses python 3.11 to avoid the bug do you remember ? :)

Reply:
Yeah, but why `matrix.python == '3.10' && '3.11' || matrix.python` ? Shouldn't `matrix.os == 'macos-latest' && matrix.python == '3.10'` be enough ?

Reply:
@odulcy-mindee syntax is a bit confusing the last && defines which value to use so

matrix.os == 'macos-latest' && matrix.python == '3.10' -> if both True use 3.11 (&& '3.11')  else use the matrix.python version

We could also used the version before so if macos-latest use always python 3.11 the change is a bit more explicit (but can be fully removed later on if we change to py 3.11 & py 3.12)

Reply:
Ah ok, thanks for the clarification